### PR TITLE
Registration app

### PR DESCRIPTION
**Pull request by Viktoriyadneprovskaya**

### CODE REVIEW

File: RegistrationApp/src/main/java/app/Main_Class.java [hunk 1]
```diff
@@ -1,6 +1,6 @@
+package app;
 
-
-public class Main {
+public class Main_Class {
```
Comment (YeaaaaH) [inline RegistrationApp/src/main/java/app/Main_Class.java:3]: why you've renamed it like this?

File: RegistrationApp/src/main/java/app/User.java [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+    public String getAddress() {
+        return address;
+    }
+
+    public void setAddress(String address) {
+        this.address = address;
+    }
+
+    public boolean getIsAvailable() {
```
Comment (YeaaaaH) [inline RegistrationApp/src/main/java/app/User.java:91]: rename it to isAvailible()

File: RegistrationApp/src/main/java/app/util/UserUtils.java [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+            return false;
+        }
+    }
+
+    public boolean validatePhoneNumber(String number) {
+        return number.matches("^[+][0-9]{12}");
+    }
+
+    //task #1
+    public List<String> showUsersLogins(List<User> users) {
```
Comment (YeaaaaH) [inline RegistrationApp/src/main/java/app/util/UserUtils.java:152]: apply correct format for streams to all methods

File: RegistrationApp/src/test/java/app/util/UserUtilsTest.java [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+        List<User> users=createUsers();
+        users.add(createUser(111,"username1"+0, "SFirstName1"+0, "Last Name1"+0+"", "someone1@gmail.com1", LocalDate.of(1900,11,10),true,"+380502456781"));
+        users.add(createUser(112,"username1"+112, "SFirstName1"+112, "Last Name1"+112, "someone1@gmail.com2", LocalDate.of(1901,11,10),true,"+380502456781"));
+        Map<Integer, LocalDate> groupedUsers = utils.findEarliestDate(users);
+        assertThat(groupedUsers.size(),is(1));
+        assertThat(groupedUsers, IsMapContaining.hasEntry(111, LocalDate.of(1900,11,10)));
+    }
+
+    @Test
+    void filterByYearSortByMonth() {
```
Comment (YeaaaaH) [inline RegistrationApp/src/test/java/app/util/UserUtilsTest.java]: you don't need to clear it

File: RegistrationApp/users.txt [hunk 5]
```diff
@@ -0,0 +1,9 @@
+0,Usr2,Avr@123Avr,Ikar,Smit,ad@gmail.com,+380666111111,true,1988-01-01
```
Comment (YeaaaaH) [inline RegistrationApp/users.txt:1]: loading user information to public is a bad idea))

File: RegistrationApp/RegistrationApp.iml [hunk 6]
```diff
@@ -1,11 +1,23 @@
 <?xml version="1.0" encoding="UTF-8"?>
```
Comment (YeaaaaH) [inline RegistrationApp/RegistrationApp.iml:1]: you don't need .iml files in git

**Review by YeaaaaH — CHANGES_REQUESTED** [review]
need to fix

File: RegistrationApp/src/main/java/app/User.java (hunk 2 above)
Comment (Viktoriyadneprovskaya) [reply RegistrationApp/src/main/java/app/User.java:91]: this is GETTER. Why do we need to rename it?

File: RegistrationApp/src/main/java/app/util/UserUtils.java (hunk 3 above)
Comment (Viktoriyadneprovskaya) [reply RegistrationApp/src/main/java/app/util/UserUtils.java:152]: fixed

File: RegistrationApp/src/test/java/app/util/UserUtilsTest.java (hunk 4 above)
Comment (Viktoriyadneprovskaya) [reply RegistrationApp/src/test/java/app/util/UserUtilsTest.java]: here is the result with removed 'testUser.clear();'
image

File: RegistrationApp/src/main/java/app/User.java (hunk 2 above)
Comment (YeaaaaH) [reply RegistrationApp/src/main/java/app/User.java:91]: for boolean types you don't need  'get' before isVARIABLE()

File: RegistrationApp/users.txt (hunk 5 above)
Comment (Viktoriyadneprovskaya) [reply RegistrationApp/users.txt:1]: Yep, but it was our split desigion. What do you suggest to do with this?

File: RegistrationApp/src/test/java/app/util/UserUtilsTest.java (hunk 4 above)
Comment (YeaaaaH) [reply RegistrationApp/src/test/java/app/util/UserUtilsTest.java]: you need to rewrite it, logic is unclear
why you create List<User> testUser = new ArrayList<>(); adding user there and test it? after that you clear it. Let's do it together on the lesson

File: RegistrationApp/users.txt (hunk 5 above)
Comment (YeaaaaH) [reply RegistrationApp/users.txt:1]: i couldn't suggest you to load users to git

**Review by YeaaaaH — APPROVED** [review]

**Pull request merged**